### ISSUE TITLE
SteamVR Launches as a ghost

### ISSUE THREAD

**Issue by heckmarr** [alvr-org/ALVR#2032]
## Description
SteamVR launches in the background, but doesn't show any windows and I have to force quit by logging out.
Previously(before this log output) I was able to get the SteamVR connected, just not to the headset, as it would
cycle through disconnected, connecting, connected, disconnected.

As well with tonight's build(2024/03/17) I was able to get error 109 twice out of steamVR, as well as a hmd not found error I haven't seen before tonight. Unfortunately I can't reproduce at the moment or I'd give you the error code.

Here is my log

If I had time I'd do this. I've only got a few hours a week to spend on this.

## General Troubleshooting
- [x] I carefully followed the instructions in the [README](https://github.com/alvr-org/ALVR/blob/master/README.md) and successfully completed the setup wizard
--I should mention that setting firewall rules still fails, but I just open ports 9943 and 9942

- [x] I read the ALVR Wikis [here](https://github.com/polygraphene/ALVR/wiki) and [here](https://github.com/alvr-org/ALVR/wiki)

### Hardware
**Note**: for Linux, an upload to the [`hw-probe`](https://linux-hardware.org/) database is preferred: `hw-probe -all -upload`
https://linux-hardware.org/?probe=95e2b24425

**CPU**:

**GPU**:

**GPU Driver Version**:

**Audio**:

### Installation
**ALVR Version**:
v21.0.0-dev00
**ALVR Settings File**:
[session.json](https://github.com/alvr-org/ALVR/files/14630554/session.json)

**SteamVR Version**:
steamvr[linux_v1.14]
**Install Type**:
- [ ] Packaged (`exe`, `deb`, `rpm`, etc)
- [ ] Portable (`zip`)
- [X] Source

**OS Name and Version** (`winver` on Windows or `grep PRETTY_NAME /etc/os-release` on most Linux distributions):
KDE neon 6.0

Thanks!

**Comment by heckmarr** [comment]
I should also mention that I love this project! I've gotten it working three or four times in the past month. Just hoping a pair of trained eyes will be able to tell where I'm going wrong.

**Comment by heckmarr** [comment]
[logoutput.log](https://github.com/alvr-org/ALVR/files/14630691/logoutput.log)
whoops, I edited my log out because I thought I forgot to sanitize, but we're all good.

**Comment by heckmarr** [comment]
okay, I found the two paths were missing, sdk and runtime, but i wasn't sure what needs to go in there, so out of curiosity I put the openvr sdk, (renamed openvr to sdk from https://github.com/ValveSoftware/openvr) and copied my sniper runtime to a folder called runtime.

This got the two errors to go away, but I have a feeling they're related to these new errors.
[logupdate00.log](https://github.com/alvr-org/ALVR/files/14643290/logupdate00.log)
Seems like the vrwebhelper going to ask for a function and it not being there.

I don't know how much of this is because I'm using alvr, or if it's because I'm using linux for SteamVR. I'm going to try installing SteamVR with no particular branch and see if it pulls in the sdk.

**Comment by heckmarr** [comment]
okay, I found the two paths were missing, sdk and runtime, but i wasn't sure what needs to go in there, so out of curiosity I put the openvr sdk, (renamed openvr to sdk from https://github.com/ValveSoftware/openvr) and copied my sniper runtime to a folder called runtime.

This got the two errors to go away, but I have a feeling they're related to these new errors.
[logupdate00.log](https://github.com/alvr-org/ALVR/files/14643318/logupdate00.log)

Seems like the vrwebhelper going to ask for a function and it not being there.

I don't know how much of this is because I'm using alvr, or if it's because I'm using linux for SteamVR. I'm going to try installing SteamVR with no particular branch and see if it pulls in the sdk.

**Comment by heckmarr** [comment]
I symlinked the 'Steamworks SDK Redist' and SteamLinuxRuntime_sniper via the ln command, and managed to get those two errors to go away, now I'm on to
`STEAM_RUNTIME_HEAVY: ./steam-runtime-heavy
./vrwebhelper: error while loading shared libraries: libcef.so: cannot open shared object file: No such file or directory
`

so I'll research what libcef.so is

**Comment by heckmarr** [comment]
Well it no longer launches as a ghost process, so that's progress.

I got the thing to spit out errors! 
![Screenshot_20240318_164718](https://github.com/alvr-org/ALVR/assets/105572511/978316c3-b3a8-4eb8-b61a-194739253608)
![Screenshot_20240318_165618](https://github.com/alvr-org/ALVR/assets/105572511/b290e386-de15-4e30-b812-59b4e9354556)
And now I'm stuck in a loop. I go to unblock alvr as usual, and it crashes with one of these errors. Then disables it and I have to re-unblock.

so kinda stymied. `libcef.so` was a dead end at the version of steam I'm using, but I'm not using any of the ones they mentioned.

**Comment by heckmarr** [comment]
ah! the `libcef.so` is the chromium embedding platform, and I recall seeing a doublefree from them recently, just not sure where.

**Comment by heckmarr** [comment]
from https://github.com/ValveSoftware/steam-for-linux/issues/8373 I ran steam with -no-cef-sandbox and I got alvr to launch and connect the server, but not the client before SteamVR claims `SteamVR has encountered a critial error` and dying.

I tried enabling Direct Mode as a toggle check, but got the same result and it had switched back, so I'm guessing that's not the issue.

At this point I wonder if I should open an issue in the SteamVR github.
